lib/main.dart

```diff
@@ -0,0 +1,431 @@
+                      'value': 'Edit topic',
+                      'title': 'Редактировать тему',
+                      'icon': const Icon(Icons.edit)
+                    },
+                  }
+                      .map(
+                        (Map<String, dynamic> item) => PopupMenuItem<String>(
+                          value: item['value'],
+                          child: ListTile(
+                            leading: item['icon'],
+                            title: Text(item['title']),
+                          ),
+                        ),
+                      )
+                      .toList())
+        ],
+      ),
+      body: feedIsEmpty
+          ? Center(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: [
+                  Stack(
+                    children: [
+                      SvgPicture.asset(
+                        'assets/todolist_background.svg',
+                      ),
+                      SvgPicture.asset(
+                        'assets/todolist.svg',
+                      ),
+                    ],
+                  ),
+                  const Flexible(
+                    child: Text(
+                      'На данный момент задачи отсутствуют',
+                      style: TextStyle(fontSize: 30),
+                    ),
+                  ),
+                ],
+              ),
+            )
+          : Center(
+              child: TaskList(
+                flag: flag,
+                taskTitles: taskTitles,
+                taskIsDone: taskIsDone,
+                taskIsFav: taskIsFav,
+                callback: callback,
+              ),
+            ),
+      floatingActionButton: FloatingActionButton(
+        tooltip: 'Добавить',
+        onPressed: addTask,
+        child: const Icon(Icons.add),
+      ),
+    );
+  }
+}
+
+class TaskList extends StatefulWidget {
+  const TaskList({
+    super.key,
+    required this.flag,
+    required this.taskTitles,
+    required this.taskIsDone,
+    required this.taskIsFav,
+    required this.callback,
+  });
+
+  final int flag;
+  final List<String> taskTitles;
+  final List<bool> taskIsDone;
+  final List<bool> taskIsFav;
+  final Function callback;
+
+  final favIcon = const Icon(Icons.star_border_purple500_outlined,
+      color: Colors.deepPurple);
+  final favIconSelected =
+      const Icon(Icons.star, color: Colors.deepPurpleAccent);
+
+  @override
+  State<TaskList> createState() => _TaskListState();
+}
+
+class _TaskListState extends State<TaskList> {
+  bool flagChecker(int index) {
+    bool isDone = widget.taskIsDone[index];
+    bool isFav = widget.taskIsFav[index];
+    return widget.flag == 0 ||
+        (widget.flag == 1 && !isDone) ||
+        (widget.flag == 2 && isFav) ||
+        (widget.flag == 3 && !isDone && isFav);
+  }
+
+  ClipRect taskBuilder(int index) {
```

Comment:
```suggestion
  Widget taskBuilder(int index) {
```
Не нужно без надобности конкретизировать